src/util/math.ts

```diff
@@ -14,7 +14,13 @@ export function accuratePow64(x: f64, y: f64): f64 {
     // This speculative optimization leads to loose precisions like 10 ** 208 != 1e208
     // or/and 10 ** -5 != 1e-5 anymore. For avoid this behaviour we are forcing exponent
     // to fractional form and compensate this afterwards.
-    if (isFinite(y) && Math.abs(y) >= 2 && Math.trunc(y) == y) {
+    if (
+      isFinite(y) &&
+      !isPowerOf2(x) &&
+      Math.abs(y) >= 2 &&
+      Math.trunc(y) == y
```

Comment:
```suggestion
      Math.trunc(y) == y &&
      !(Math.trunc(x) === x && y > 0 && Math.log2(Math.abs(x)) * y < 53)
```

this should help for f64(3**7)

Reply:
the idea is to check if  Number.isSafeInteger(a**b)

Reply:
@MaxGraey Is this comment something to address?

Reply:
All this checks become a really complex thing. It might be worth just compiling the pow function from WebAssembly and always using it instead try to fix browser's one

Reply:
Would this imply that we'd remove `accuratePow64` as an optimization and always call the full impl?

Reply:
Or is it rather that we need, say, a asm.js version of the musl impl and use it as a new `accuratePow64`?

Reply:
Yes. I guess it make sense to do this in another PR. So we just need to make a js wrapper over `Math.pow` of assemblyscript compiled to wasm and use it inside compiler as well.